app/src/main/java/com/kickstarter/viewmodels/CommentsViewModel.java

```diff
@@ -39,311 +37,437 @@
+
+      projectOrUpdate
+              .filter(Either::isLeft)
+              .map(Either::left)
+              .compose(takeWhen(postedComment))
+              .compose(bindToLifecycle())
+              .subscribe(this.koala::trackProjectCommentCreate);
+    }
+
+    private @NonNull
+    Observable<Comment> postComment(final @NonNull Either<Project, Update> projectOrUpdate, final @NonNull String body) {
+      return projectOrUpdate.either(
+              p -> this.client.postComment(p, body),
+              u -> this.client.postComment(u, body)
+      );
+    }
+
+    private final PublishSubject<String> commentBodyChanged = PublishSubject.create();
+    private final PublishSubject<Void> commentButtonClicked = PublishSubject.create();
+    private final PublishSubject<Void> commentDialogDismissed = PublishSubject.create();
+    private final PublishSubject<Boolean> commentIsPosting = PublishSubject.create();
+    private final PublishSubject<Void> loginSuccess = PublishSubject.create();
+    private final PublishSubject<Void> nextPage = PublishSubject.create();
+    private final PublishSubject<Void> postCommentClicked = PublishSubject.create();
+    private final PublishSubject<Void> refresh = PublishSubject.create();
+
+    private final BehaviorSubject<CommentsData> commentsData = BehaviorSubject.create();
+    private final BehaviorSubject<String> currentCommentBody = BehaviorSubject.create();
+    private final BehaviorSubject<Void> dismissCommentDialog = BehaviorSubject.create();
+    private final BehaviorSubject<Boolean> enablePostButton = BehaviorSubject.create();
+    private final BehaviorSubject<Boolean> isFetchingComments = BehaviorSubject.create();
+    private final BehaviorSubject<Boolean> commentButtonHidden = BehaviorSubject.create();
+    private final BehaviorSubject<Pair<Project, Boolean>> showCommentDialog = BehaviorSubject.create();
+    private final PublishSubject<Void> showCommentPostedToast = PublishSubject.create();
+    private final PublishSubject<ErrorEnvelope> showPostCommentErrorToast = PublishSubject.create();
+
+    public final Inputs inputs = this;
+    public final Outputs outputs = this;
+
+    @Override
+    public void commentBodyChanged(final @NonNull String string) {
+      this.commentBodyChanged.onNext(string);
+    }
+
+    @Override
+    public void commentButtonClicked() {
+      this.commentButtonClicked.onNext(null);
+    }
+
+    @Override
+    public void commentDialogDismissed() {
+      this.commentDialogDismissed.onNext(null);
+    }
+
+    @Override
+    public void loginSuccess() {
+      this.loginSuccess.onNext(null);
+    }
+
+    @Override
+    public void nextPage() {
+      this.nextPage.onNext(null);
+    }
+
+    @Override
+    public void postCommentClicked() {
+      this.postCommentClicked.onNext(null);
+    }
+
+    @Override
+    public void refresh() {
+      this.refresh.onNext(null);
+    }
+
+    @Override
```

Comment:
should format as 
```java
@Override public @NonNull Observable<CommentsData> commentsData() {
```
since these are all essentially boilerplate setter functions we can keep them concise and at the bottom of the file to keep the files shorter!

Reply:
Thanks for pointing this out. I just realized how these methods are formatted. I have no idea how they got formatted like that :/